textdata_reader.py

```diff
@@ -0,0 +1,181 @@
+# limitations under the License.
+# pylint: disable=attribute-defined-outside-init
+"""
+mimic-cxr 2.0.0 medical report preprocessor for the text that does extraction
+of findings and impression part of the text. Tokenize the sentence, lowercase
+all the characters and remove the word contains non-alphabetic characters
+"""
+import argparse
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+from forte.data.readers import PlainTextReader
+from forte.data.span import Span
+from forte.pipeline import Pipeline
+from forte.processors.nltk_processors import NLTKWordTokenizer
+from forte.processors.lowercaser_processor import LowerCaserProcessor
+from mimic.onto.mimic_ontology import Impression, Findings, FilePath
+from forte.data.multi_pack import MultiPack
+from forte.processors.base import MultiPackProcessor
+from forte.data.caster import MultiPackBoxer
+from ft.onto.base_ontology import Token
+from forte.processors.writers import PackNameJsonPackWriter
+from forte.data.selector import NameMatchSelector
+
+
+class FindingsExtractor(PackProcessor):
+    r"""A wrapper of fingings extractor.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def _process(self, input_pack: DataPack):
+        findings_ind = input_pack.text.find("FINDINGS")
+        impression_ind = input_pack.text.find("IMPRESSION")
+        if findings_ind == -1:
+            begin = 0
+            end = 0
+        else:
+            begin = findings_ind + len("FINDINGS") + 1
+            if impression_ind != -1 and findings_ind < impression_ind:
+                end = impression_ind - 1
+            else:
+                end = len(input_pack.text)
+        findings = Findings(input_pack, begin, end)
+        findings.has_content = (findings_ind != -1)
+
+
+class ImpressionExtractor(PackProcessor):
+    r"""A wrapper of impression extractor.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def _process(self, input_pack: DataPack):
+        findings_ind = input_pack.text.find("FINDINGS")
+        impression_ind = input_pack.text.find("IMPRESSION")
+        if impression_ind == -1:
+            begin = 0
+            end = 0
+        else:
+            begin = impression_ind + len("IMPRESSION") + 1
+            if findings_ind != -1 and impression_ind < findings_ind:
+                end = findings_ind - 1
+            else:
+                end = len(input_pack.text)
+        impression = Impression(input_pack, begin, end)
+        impression.has_content = (impression_ind != -1)
+
+
+class FilePathGetter(PackProcessor):
+    r"""A wrapper to get the file path hierarchy of the current file
+    """
+    def __init__(self):
+        super().__init__()
+
+    def _process(self, input_pack: DataPack):
+        filepath = FilePath(input_pack)
+        if input_pack.pack_name is not None:
+            filepath.img_study_path = input_pack.pack_name
```

Comment:
This processor seems a bit trivial. You can directly do this inside the reader.

Reply:
Also, it looks like this will not be part of the `result` pack, because this is called before that pack is created.